src/Controller/Plugin/Forward.php

```diff
@@ -125,7 +126,13 @@ public function dispatch($name, array $params = null)
         // Allow passing parameters to seed the RouteMatch with & copy matched route name
         if ($params !== null) {
             $routeMatch = new RouteMatch($params);
-            $routeMatch->setMatchedRouteName($event->getRouteMatch()->getMatchedRouteName());
+            $eventRouteMatch = $event->getRouteMatch();
+
+            if (! $eventRouteMatch) {
```

Comment:
You previously used `null !==`, while here you used `!`: pick one of the two approaches please :)